rmw_opensplice_cpp/src/demangle.cpp

```diff
@@ -36,19 +37,21 @@ _demangle_if_ros_topic(const std::string & topic_name)
 std::string
 _demangle_if_ros_type(const std::string & dds_type_string)
 {
-  std::string substring = "::msg::dds_::";
+  std::string substring = "dds_::";
   size_t substring_position = dds_type_string.find(substring);
   if (
-    dds_type_string[dds_type_string.size() - 1] == '_' &&
-    substring_position != std::string::npos)
+    dds_type_string[dds_type_string.size() - 1] != '_' ||
+    substring_position == std::string::npos)
   {
-    std::string pkg = dds_type_string.substr(0, substring_position);
-    size_t start = substring_position + substring.size();
-    std::string type_name = dds_type_string.substr(start, dds_type_string.length() - 1 - start);
-    return pkg + "/" + type_name;
+    // not a ROS type
+    return dds_type_string;
   }
-  // not a ROS type
-  return dds_type_string;
+
+  std::string type_namespace = dds_type_string.substr(0, substring_position);
+  type_namespace = std::regex_replace(type_namespace, std::regex("::"), "/");
+  size_t start = substring_position + substring.size();
+  std::string type_name = dds_type_string.substr(start, dds_type_string.length() - 1 - start);
+  return type_namespace + type_name;
```

Comment:
Inverting the logic makes the diff more difficult to review. What was the reason to do so?

Reply:
Consistency. The logic now matches the other `rmw` implementations. I can revert it or break it into two commits if you prefer.

Reply:
No need it - the change looks good. I just wanted to check.